lineage/BaumWelch.py

```diff
@@ -86,9 +87,10 @@ def do_M_step(tHMMobj, MSD, betas, gammas):
 
     if tHMMobj[0].estimate.fE is None:
         assert tHMMobj[0].fE is None
-        for idx, tt in enumerate(tHMMobj):
-            do_M_E_step(tt, gammas[idx])
-
+        if len(tHMMobj) == 1: # means it only performs calculation on one condition at a time.
+            do_M_E_step(tHMMobj[0], gammas[0])
+        else: # means it performs the calculations on several concentrations at once.
```

Comment:
Does this mean this will be applied in Fig. 10? Do the emissions we generate in that figure satisfy our constraint here? We'll also need to adjust the degrees of freedom provided to AIC (not here but make an issue).

Reply:
Yes, I hope it can help with solving the issue that we had with figure 10 (every time it looked different).
I will make the issue right now.

Reply:
I have to look into the emissions, but I think the data is the same, so should be the emissions.

Reply:
I will take a closer look at this in another PR.